docs/dev/swagger-source.json

```diff
@@ -1287,6 +1291,336 @@
+						"name": "contributorId",
+						"in": "path",
+						"description": "Contributor ID",
+						"required": true,
+						"type": "integer"
+					}
+				],
+				"responses": {
+					"200": {
+						"x-summary": "Success",
+						"description": "The contributor details",
+						"schema": {
+							"$ref": "#/definitions/Contributor"
+						}
+					},
+					"403": {
+						"x-summary": "Publication invalid",
+						"description": "The requested publication is not part of the requested submission. It could be also because the publication is not editable by the requested user",
+						"schema": {
+							"$ref": "#/definitions/ErrorResponse"
+						}
+					},
+					"404": {
+						"x-summary": "Request Not Found",
+						"description": "The requested submission or publication or author could not be found.",
+						"schema": {
+							"$ref": "#/definitions/ErrorResponse"
+						}
+					}
+				}
+			},
+			"put": {
+				"tags": [
+					"Submissions / Publications / Contributors"
+				],
+				"summary": "Edit a Contributor.",
+				"parameters": [
+					{
+						"name": "submissionId",
+						"in": "path",
+						"description": "Submission ID",
+						"required": true,
+						"type": "integer"
+					},
+					{
+						"name": "publicationId",
+						"in": "path",
+						"description": "Publication ID",
+						"required": true,
+						"type": "integer"
+					},
+					{
+						"name": "contributorId",
+						"in": "path",
+						"description": "Contributor ID",
+						"required": true,
+						"type": "integer"
+					},
+					{
+						"description": "The details about the contributor that you want to edit. All params are optional. When a param is omitted, the existing value of that param will not be modified.",
+						"in": "body",
+						"schema": {
+							"$ref": "#/definitions/ContributorEditable"
+						}
+					}
+				],
+				"responses": {
+					"200": {
+						"x-summary": "Success",
+						"description": "The contributor that was just edited with the new values after the update has occurred.",
+						"schema": {
+							"$ref": "#/definitions/Contributor"
+						}
+					},
+					"400": {
+						"x-summary": "Request Invalid",
+						"description": "The values you sent with the request did not pass validation. The response will contain information about which parameters failed validation.",
+						"schema": {
+							"$ref": "#/definitions/InvalidParametersResponse"
+						}
+					},
+					"403": {
+						"x-summary": "Request Unauthorized",
+						"description": "This error indicates one of the following:\n* The publication you want to edit does not belong to this submission.\n* This publication is published and can not be edited. You must create and publish a new version.\n* You do not have permission to edit this publication.\n* You are not allowed to modify the `status` of a publication. You must use the `/submissions/{submissionId}/publications/{publicationId}/publish` and `.../unpublish` endpoints to modify the status of a publication.",
```

Comment:
I think this needs to be updated. I'd suggest the following:

* This contributor is assigned to a publication that does not belong to this submission.
* This contributor is assigned to a publication that has already been published. You must create a new version of the publication to modify contributor details.
* You do not have permission to edit contributors for this publication.

Reply:
Changed